### PR TITLE
MINOR: Add Timed wait to SslTransportLayerTest.testNetworkThreadTimeRecorded

### PR DESCRIPTION
**Pull request by omkreddy**

### CODE REVIEW

**Comment by omkreddy** [comment]
@rajinisivaram  SslTransportLayerTest.testNetworkThreadTimeRecorded test is consistently getting stuck in infinite while loop (check below stack trace) on my machine. looks like some timing issues.  spent some time to debug the issue, but no success.

For now, i changed the while loop to timed wait loop. Pls take a look.

```
"Test worker" #12 prio=5 os_prio=31 tid=0x00007f7fecbb9000 nid=0x5a03 runnable [0x00007000015d2000]
   java.lang.Thread.State: RUNNABLE
        at sun.nio.ch.KQueueArrayWrapper.kevent0(Native Method)
        at sun.nio.ch.KQueueArrayWrapper.poll(KQueueArrayWrapper.java:198)
        at sun.nio.ch.KQueueSelectorImpl.doSelect(KQueueSelectorImpl.java:117)
        at sun.nio.ch.SelectorImpl.lockAndDoSelect(SelectorImpl.java:86)
        - locked <0x00000007969a1b98> (a sun.nio.ch.Util$2)
        - locked <0x00000007969a1b88> (a java.util.Collections$UnmodifiableSet)
        - locked <0x00000007969a1a68> (a sun.nio.ch.KQueueSelectorImpl)
        at sun.nio.ch.SelectorImpl.select(SelectorImpl.java:97)
        at org.apache.kafka.common.network.Selector.select(Selector.java:684)
        at org.apache.kafka.common.network.Selector.poll(Selector.java:408)
        at org.apache.kafka.common.network.SslTransportLayerTest.testNetworkThreadTimeRecorded(SslTransportLayerTest.java:665)
```

**Comment by omkreddy** [comment]
retest this please.

File: clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java [hunk 1]
```diff
@@ -661,10 +661,20 @@ public void testNetworkThreadTimeRecorded() throws Exception {
...
+        TestUtils.waitForCondition(new TestCondition() {
+            @Override
+            public boolean conditionMet() {
+                try {
+                    selector.poll(100L);
+                    assertEquals(0, selector.numStagedReceives(channel));
+                } catch (IOException e) {
+                    return false;
+                }
+                return selector.completedReceives().size() > 0;
```
Comment (rajinisivaram) [inline clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java]: Could use !`isEmpty()` instead of `.size()`?

**Review by rajinisivaram — COMMENTED** [review]
@omkreddy Thanks for the PR. Left a minor comment. So does this fail consistently with the PR? If so, can you run with debug logging and raise a JIRA with the logs? Thanks.

File: clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java (hunk 1 above)
Comment (omkreddy) [reply clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java]: @rajinisivaram  Thanks for the review.  test is failing most of the times.   I will create JIRA with relevant logs.

**Review by rajinisivaram — APPROVED** [review]
@omkreddy Thanks for the update, LGTM.

**Comment by rajinisivaram** [comment]
Test failures are unrelated, merging to trunk.

**Pull request merged**